### PR TITLE
Svdesu/ch95/edit a circuit

### PR DESCRIPTION
**Pull request by SVD6**
Story: https://app.clubhouse.io/chrono/story/95/edit-a-circuit

Basically you can just edit a circuit now. I did however add some toasts. That's it.

### CODE REVIEW

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreateActivity.kt [hunk 1]
```diff
@@ -24,17 +24,25 @@ import com.google.android.material.dialog.MaterialAlertDialogBuilder
...
+    private var editPosition: Int = -1
     private lateinit var iconNames: TypedArray
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         PreferenceManager.with(this)
         bind = DataBindingUtil.setContentView(this, R.layout.activity_circuit_create)
 
+        if (intent.getBooleanExtra("isEdit", false)) {
+            isEdit = true
```
Comment (harigovind1998) [inline app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreateActivity.kt:41]: So isEdit is only passed through to the intent if edit, else "isEdit" isnt present in the intent? I think you should check isEdit == true here. Just to be safe.

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitDashboardFrag.kt [hunk 2]
```diff
@@ -73,12 +76,29 @@ class CircuitDashboardFrag : Fragment() {
...
         super.onActivityResult(requestCode, resultCode, data)
-        if (requestCode == 10001 && resultCode == Activity.RESULT_OK) {
-            loadData()
-        }
-        if (requestCode == 10002 && resultCode == Activity.RESULT_OK) {
-            // Ideal spot to ask for a rating after a threshold of timers have been run
-            Log.i("circuit_activity", "Completed a circuit.")
+        if (resultCode == Activity.RESULT_OK) {
+            when (requestCode) {
+                10001 -> {
```
Comment (harigovind1998) [inline app/src/main/java/ca/chronofit/chrono/circuit/CircuitDashboardFrag.kt]: would suggest creating a variable so it doesnt seem arbritary

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitDashboardFrag.kt [hunk 3]
```diff
@@ -151,11 +171,11 @@ class CircuitDashboardFrag : Fragment() {
...
-            Toast.makeText(
-                requireContext(),
-                "\uD83D\uDEE0\uFE0F Edit circuit coming soon!! \uD83D\uDEE0\uFE0F",
-                Toast.LENGTH_SHORT
-            ).show()
+            val intent = Intent(requireContext(), CircuitCreateActivity::class.java)
+            intent.putExtra("isEdit", true)
+            intent.putExtra("circuitPosition", position)
+            dialog.dismiss()
+            startActivityForResult(intent, 10003)
```
Comment (harigovind1998) [inline app/src/main/java/ca/chronofit/chrono/circuit/CircuitDashboardFrag.kt]: 10003 is being used in multiple places as well.

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreateActivity.kt (hunk 1 above)
Comment (SVD6) [reply app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreateActivity.kt:41]: There's a default value there that's why I'm not bothering. So if the intent isn't present it just reverts to false.

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitDashboardFrag.kt (hunk 3 above)
Comment (SVD6) [reply app/src/main/java/ca/chronofit/chrono/circuit/CircuitDashboardFrag.kt]: Yeah that's starting the activity for code 10003, the other comment you made is where I'm getting the same activity's result. Yeah I'll quickly make some constants

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitDashboardFrag.kt (hunk 2 above)
Comment (SVD6) [reply app/src/main/java/ca/chronofit/chrono/circuit/CircuitDashboardFrag.kt]: done that now

**Pull request merged**